### ISSUE TITLE
efficient-det training on custom data set - where to start? 

### ISSUE THREAD

**Issue by Suzan009** [google/automl#1127]
I want to try efficientdet. But I don't know where to start. Are there any end-to-end tutorials regarding this? 

- how should I prepare my data? For example: what model would expect the data format (image and target)
- how should I prepare my model and start training with the data 
- how should I evaluate 
- how should I do inference 
- how should I fine-tune the trained model? 
- and what the best practice, tips, and tricks to get better performance?

Also, a few query

- Can I train square and non-square input images? 
- Is it possible if I want to change or modify any intermediate layer of efficient-det?

**Comment by Suzan009** [comment]
@fsx950223

**Comment by fsx950223** [comment]
https://github.com/google/automl/blob/master/efficientdet/tutorial.ipynb
You could use non-square input images as train dataset, but I can't train a SOTA model when I use non-square input images.
You could tune num_scales min_level max_level to change intermediate layers.

**Comment by Suzan009** [comment]
@fsx950223 The above tutorials don't use [`tf2`](https://github.com/google/automl/tree/master/efficientdet/tf2) but use old tf1, I think, isn't it?

**Comment by Suzan009** [comment]
tutorials mention [here ](https://github.com/google/automl/tree/master/efficientdet/tf2#efficientdet )is broken I think

**Comment by fsx950223** [comment]
https://colab.research.google.com/drive/13_XLWNDYV9IAZMAmu8pxmdTNbwaoPI-a?usp=sharing

**Comment by fsx950223** [comment]
You could reference readme under tf2.

**Comment by Suzan009** [comment]
Could you please re-check the above colab link? When I try to open it , it says 'Sorry, the file you have requested does not exist.' :(

**Comment by fsx950223** [comment]
> Could you please re-check the above colab link? When I try to open it , it says 'Sorry, the file you have requested does not exist.' :( 
> 
> ![image](https://user-images.githubusercontent.com/45315076/144744786-514106d0-7b7b-4fb4-8079-aadeb093bc13.png)
> 

It's not available anymore.

**Comment by Suzan009** [comment]
@fsx950223 
I've some queries regarding the config file. For example, 

(1). [Input preprocessing parameters. ](https://github.com/google/automl/blob/39c39e5b635f73e86082bff3deddefecfbc0a123/efficientdet/hparams_config.py#L180-L189) - Here are some augmentation parameters. I wonder how the model would differentiate the train set and validation set? Isn't augmentation should only apply to the train set? But here in the config file, there is no explicit mention of the parameter for a train set and validation set. 

(2). [Segmentation class. ](https://github.com/google/automl/blob/39c39e5b635f73e86082bff3deddefecfbc0a123/efficientdet/hparams_config.py#L194) 

For my object detection task, I don't need to make any segmentation outcome. Should I leave all the segmentation settings by default. By default, it sets 3 for the number of segmentation classes. Should I change it to 0 or skip the settings. 


(3). [Post-processing. NMS](https://github.com/google/automl/blob/39c39e5b635f73e86082bff3deddefecfbc0a123/efficientdet/hparams_config.py#L259-L267)

Here are only NMS pp. But I found the wbf.py. So, if I want to use it, how should I do that?

**Comment by fsx950223** [comment]
Augmentation parameters are not available in val set.
Default config doesn't contain segmentation branch.

**Comment by Suzan009** [comment]
@fsx950223 thanks, understood. 
I went through the efficientdet code-bases. Is it possible to work on the model definition separately without data config set up and others? I want the efficientdet model definition separately. 

I made data -> preprocess it -> (use this repo's efficient-det) -> post-processes.  I don't need the official data processing pipelines. Is it possible to work on only the model arch? Also without any other stuff like tfmot or tflite?

**Comment by fsx950223** [comment]
You could take a look at tf2/efficientdet_keras.py

**Comment by Suzan009** [comment]
I tried actually. But [this line](https://github.com/google/automl/blob/39c39e5b635f73e86082bff3deddefecfbc0a123/efficientdet/tf2/efficientdet_keras.py#L826) took me to many places and later I realize it's a bit complex for me. I've worked with classification, semantic-segmentation. For that, it's a bit complex for me now. Is there any similar (but simpler) efficient-det implementation? How about tf-hub? Is it trainable out of the box? Sorry for bothering, a bit confused.
